spec/lib/telephony/otp_sender_spec.rb

```diff
@@ -255,4 +255,100 @@
       end
     end
   end
+
+  describe '#authentication_message' do
+    let(:sender) do
+      sender = Telephony::OtpSender.new(
+        to: '+18888675309',
+        otp: 'ABC123',
+        channel: 'sms',
+        expiration: TwoFactorAuthenticatable::DIRECT_OTP_VALID_FOR_MINUTES,
+        domain: 'secure.login.gov',
+        country_code: 'US',
+      )
+    end
+
+    context 'sms' do
+      context 'English' do
+        it 'does not contain any non-GSM characters and is less than or equal to 160 characters' do
+          message = sender.authentication_message
+          expect(Telephony.sms_parts(message)).to eq 1
+          expect(message.chars.all? { |x| Telephony::GSM_CHARACTERS.include?(x) }).to eq true
+        end
+      end
+
+      # The Spanish-language translation currently includes the 'ó', character, which is not
+      # in the GSM 03.38 character set
+      context 'Spanish' do
+        it 'is sent in three parts' do
+          I18n.locale = :es
+
+          message = sender.authentication_message
+          expect(Telephony.sms_parts(message)).to eq 3
+        ensure
+          I18n.locale = :en
+        end
+      end
+
+      context 'French' do
+        it 'does not contain any non-GSM characters and is less than or equal to 160 characters' do
+          I18n.locale = :fr
+
+          message = sender.authentication_message
+          expect(Telephony.sms_parts(message)).to eq 1
+          expect(message.chars.all? { |x| Telephony::GSM_CHARACTERS.include?(x) }).to eq true
+        ensure
+          I18n.locale = :en
+        end
+      end
+    end
+  end
+
+  describe '#confirmation_message' do
+    let(:sender) do
+      sender = Telephony::OtpSender.new(
+        to: '+18888675309',
+        otp: 'ABC123',
+        channel: 'sms',
+        expiration: TwoFactorAuthenticatable::DIRECT_OTP_VALID_FOR_MINUTES,
+        domain: 'secure.login.gov',
+        country_code: 'US',
+      )
+    end
+
+    context 'sms' do
+      context 'English' do
+        it 'does not contain any non-GSM characters and is sent in one part' do
+          message = sender.confirmation_message
+          expect(Telephony.sms_parts(message)).to eq 1
+          expect(message.chars.all? { |x| Telephony::GSM_CHARACTERS.include?(x) }).to eq true
```

Comment:
WDYT of making a custom matcher for this? `expect(message).to be_all_gsm_characters`

Reply:
Oren had a similar suggestion [here](https://github.com/18F/identity-idp/pull/5752#discussion_r775622338) that I added in https://github.com/18F/identity-idp/commit/61bd390839f61525ce1946b2d55ce3c6e2e41140